### ISSUE TITLE
Combining Multiple Metrics into Same Graph Dynamically

### ISSUE THREAD

**Issue by iperry-indigex** [Icinga/icingaweb2-module-graphite#276]
We have checks that provide performance data in a dynamic way, e.g. differing numbers of sensors. We'd like to combine all metrics of a type (e.g. all temps) into one graph dynamically. I can combine them if I explicitly define each metric, but we need them to be dynamic. There's no documentation on this issue and the forums have not provided help.

## Your Environment
* Module version (System - About): 1.2.0
* Icinga Web 2 version and modules (System - About): 2.10.1
* Icinga 2 version (`icinga2 --version`): r2.13.3-1
* Graphite (carbon, web, whisper) version: 1.1.10
* Operating System and version: Ubuntu 20.04.4
* Webserver, PHP versions: Apache 2.4.41, PHP 7.4.3

**Comment by nilmerg** [comment]
Hi,

have you checked the templates chapter? (https://icinga.com/docs/icinga-module-for-graphite/latest/doc/04-Templates/#templates-structure-metric-filters)

Also, take a look at the disk default template.

**Comment by iperry-indigex** [comment]
> Hi,
> 
> have you checked the templates chapter? (https://icinga.com/docs/icinga-module-for-graphite/latest/doc/04-Templates/#templates-structure-metric-filters)
> 
> Also, take a look at the disk default template.

Yes, read through it multiple times. I took a look at the disk template and the issue I'm experiencing is different. I'm able to dynamically define things like that. Allow me to attempt to explain a little better.

We have a device that has 24 temperature probes. As of right now, all of the temperatures for those probes appear on separate graphs. I'd prefer they all appeared on the same graph. However, instead of defining temp1, temp2, etc... we have devices with variable numbers of temperature sensors, and I would like to take n number of temperatures, dynamically determined, and put them all on one graph. I can do it in the graphite interface by doing perfdata.temp_*.value, but that doesn't appear to work in the .ini.

**Comment by nilmerg** [comment]
Ah, got it. Yes, the wildcard support here only allows to dynamically render multiple graphs out of a single template. Lines/Bars/Slices however cannot be rendered dynamically.

**Comment by iperry-indigex** [comment]
> Ah, got it. Yes, the wildcard support here only allows to dynamically render multiple graphs out of a single template. Lines/Bars/Slices however cannot be rendered dynamically.

Noted. This is something PNP4Nagios could do and it's something graphite can do, so I figured there'd be a way to get icinga for graphite to do it. Thank you.

**Comment by PeterLustig1337** [comment]
> > Ah, got it. Yes, the wildcard support here only allows to dynamically render multiple graphs out of a single template. Lines/Bars/Slices however cannot be rendered dynamically.
> 
> Noted. This is something PNP4Nagios could do and it's something graphite can do, so I figured there'd be a way to get icinga for graphite to do it. Thank you.

How did you do it?
Did you manage to get a different coloring for all perfdata_label dynamically?
Or did you just combined them with one color?
